File: 283-MoveZeroes/283-MoveZeroes.py
# Last updated: 8/16/2026, 9:50:42 PM
class Solution:
    def moveZeroes(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """

        # non-zero counts
        # make nums[0~cnt] to nonZeros, and left as 0
        cnt = 0

        for i in range(len(nums)):
            if nums[i] != 0:
                nums[cnt] = nums[i]
                cnt += 1
        
        for i in range(cnt, len(nums)):
            nums[i] = 0
        
        return


        

File: 283-MoveZeroes/283-MoveZeroes.java
// Last updated: 8/16/2026, 9:50:43 PM
class Solution {
    public void moveZeroes(int[] nums) {
        /*
        0,1,0,3,12
        1,3,12,0,0

        int nonZeos = records # of non-0
        make nums[0~nonZeros] to nonZeros, and left as 0
        */
        int nonZeros = 0;
        for(int num: nums) {
            if(num != 0) {
                nums[nonZeros] = num;
                nonZeros ++;
            }
        }

        if(nonZeros > 0) {
            for(int i = nonZeros; i < nums.length; i ++) {
                nums[i] = 0;
            }
        }
    }
}